PLAT-24017 Can use configuration from Weblink Builder

gbannerman opened the pull request:
This change adds the ability to specify the `configuration` and `product_route` arguments for the `[administrate-widget]` shortcode. The `configuration` argument will allow users to take a JSON configuration string from WebLink Builder and use it directly in the shortcode. This will override any other arguments given to the shortcode. The `product_route` argument can be used to specify a link builder function for the Catalogue widget using that `product_route` and the products name in kebab-case (e.g. if the `product_route` is `/foo`, Course Bar would be available at `https://mywebsite.test/foo/course-bar`). If the `configuration` argument defines the `links` object, then the `product_route` is ignored.

Both of these arguments can be configured using controls in the Weblink2 Widget block editor UI, which also allows us to show help messages to users explaining what each control does.

https://administrate.atlassian.net/browse/PLAT-24017

gbannerman committed:
PLAT-24017 Add dockerized setup for development

This commit adds a dockerized wordpress setup to the project so that
engineers can quickly and easily set up the project to begin development
.

gbannerman committed:
PLAT-24017 Allow `configuration` shortcode argument for widgets

This commit adds the ability to specify the `configuration` argument to
the `[administrate-widget]` shortcode. This will allows users to take a
JSON configuration string from WebLink Builder and add it directly to
the shortcode usage. This will override any other arguments given to the
shortcode.

gbannerman committed:
PLAT-24017 Automatically add links for catalogue widget

This commit modifies the `[administrate-widget]` shortcode so that if
the `product_route` argument is passed to it, this will be used to
specify a link builder function for the Catalogue widget using that
`product_route` and the products name in kebab-case. E.g. if the
`product_route` is `/foo`, Course Bar would be available at
`https://mywebsite.test/foo/course-bar`. If the `configuration` argument
defines the `links` object, then the `product_route` is ignored.

gbannerman committed:
PLAT-24017 Add block controls for configuration and product_route

This commit adds controls for the `configuration` and `product_route`
attributes of the Weblink2 Widget block. This allows users to configure
these using the block editor UI and also allows us to show help
messages to them explaining what each control does.

evalexpr commented on docker/wordpress/Dockerfile:
```diff
@@ -0,0 +1,10 @@
+FROM wordpress:5.8.3
+
+RUN apt update && apt install -y libxml2-dev
+RUN docker-php-ext-install soap
```
Did we verify if we needed these?

gbannerman replied:
Nope. I can verify that today.

gbannerman replied:
Yeah they don't appear to be necessary, so I've removed them and amended the commit which added the Dockerfile.

gbannerman committed:
PLAT-24017 Increase plugin version to `1.6.2`

gbannerman committed:
PLAT-24017 Update product route logic based on feedback

The new `product_route` shortcode property that we added has been
updated based on feedback. The two changes are:
1. The route no longer requires a preceding slash, e.g. it was expecting
something like `/courses` but now it just expects `courses`.
2. The function to turn the product name into kebab case has been
changed to mimic the behaviour of Wordpress's `sanitize_title` fucntion.
This is what generates the URL on the server side. We cannot get a
perfect javascript implementation for it, but this should do for most
scenarios.
